src/OperatorMsg.chpl

```diff
@@ -926,62 +999,142 @@ module OperatorMsg
+                        omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                        return new MsgTuple(errorMsg, MsgType.ERROR);
+                    }
+                }
+            }
+            when (DType.UInt64, DType.Float64) {
+                var errorMsg = notImplementedError(pn,left.dtype,op,dtype);
+                omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+            when (DType.UInt64, DType.Bool) {
+                var l = toSymEntry(left, uint);
+                var val = value.getBoolValue();
+                select op {
+                    when "+=" {l.a += val:uint;}
+                    when "-=" {l.a -= val:uint;}
+                    when "*=" {l.a *= val:uint;}
+                    otherwise {
+                        var errorMsg = notImplementedError(pn,left.dtype,op,dtype);
+                        omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                        return new MsgTuple(errorMsg, MsgType.ERROR);
+                    }
+                }
             }
             when (DType.Float64, DType.Int64) {
                 var l = toSymEntry(left,real);
                 var val = value.getIntValue();
-                select op
-                {
+                select op {
                     when "+=" {l.a += val;}
                     when "-=" {l.a -= val;}
                     when "*=" {l.a *= val;}
                     when "/=" {l.a /= val:real;} //truediv
                     when "//=" { //floordiv
-                        if val != 0 {l.a = floor(l.a / val);} else {l.a = NAN;}
+                        ref la = l.a;
+                        [li in la] li = floorDivisionHelper(li, val);
                     }
                     when "**=" { l.a **= val; }
                     otherwise {
                         var errorMsg = notImplementedError(pn,left.dtype,op,dtype);
                         omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                        return new MsgTuple(errorMsg, MsgType.ERROR);                          
+                        return new MsgTuple(errorMsg, MsgType.ERROR);
+                    }
+                }
+            }
+            when (DType.Float64, DType.UInt64) {
+                var l = toSymEntry(left,real);
+                var val = value.getUIntValue();
+                select op {
+                    when "+=" { l.a += val; }
+                    when "-=" { l.a -= val; }
+                    when "*=" { l.a *= val; }
+                    when "//=" {
+                        ref la = l.a;
+                        [li in la] li = floorDivisionHelper(li, val);
+                    }//floordiv
+                    when "%=" {
+                        if val != 0 {l.a %= val;} else {l.a = 0;}
```

Comment:
@Ethan-DeBandi99 : With a quick look, this PR seems to have broken our smoke testing with:

```
$CHPL_HOME/test/studies/arkouda/arkouda//src/OperatorMsg.chpl:954: In function 'opeqvsMsg':
$CHPL_HOME/test/studies/arkouda/arkouda//src/OperatorMsg.chpl:1109: error: unresolved call '%([domain(1,int(64),false)] real(64), uint(64))'
$CHPL_HOME/modules/internal/ChapelBase.chpl:2211: note: this candidate did not match: %(a: uint(64), param b: uint(64))
$CHPL_HOME/modules/internal/ChapelBase.chpl:2012: note: because actual argument #1 with type '[domain(1,int(64),false)] real(64)'
$CHPL_HOME/modules/internal/ChapelBase.chpl:2211: note: is passed to formal 'a: uint(64)'
$CHPL_HOME/test/studies/arkouda/arkouda//src/OperatorMsg.chpl:1109: note: other candidates are:
$CHPL_HOME/modules/internal/ChapelBase.chpl:2222: note:   %(a: int(64), param b: int(64))
$CHPL_HOME/modules/internal/ChapelBase.chpl:400: note:   %(a: int(8), b: int(8))
note: and 23 other candidates, use --print-all-candidates to see them
Makefile:321: recipe for target 'arkouda_server' failed
```

are you not seeing the same behavior?

Reply:
CI testing with Chapel 1.28 was clean, so I suspect this is from a change to chapel main, which is what our smoke testing uses (guessing something related to preferring instantiations over converting or something)

Reply:
The Arkouda CI only tests Chapel 1.28.0 and 1.29.0 currently, where I am able to reproduce this on `main`, but not one of those older releases.

Reply:
@pierce314159 internally we're trying to figure out if Chapel intends to support `%`/`%=` on reals. According to our spec, we don't yet, but previously the implementation supported `%=` (perhaps accidentally)

However, I'm wondering if you actually need/want to support it yourself here. It looks like you added support for `Float64 %= UInt64`, but I don't think this is supported for any other float parings (`Float64 %= Int64` or `Float64 %= Float64`) so I'm wondering if this was an accidental addition and not needed.

Reply:
FWIW, it looks like I may have added the %= overload on reals accidentally some time ago—at least, I don't see any real justification for it.  Also, I checked and C doesn't support `%` or `%=` on doubles (i.e., I'm not just forgetting some advanced math).  So I don't currently think Chapel should be supporting % on reals, which would suggest avoiding reliance on it in Arkouda (i.e., I'm also interested in the answer to Elliot's question about whether its use here was intentional).

I'm doing a check to see whether any other Chapel code is relying on that `%=` overload on reals...